gem/backends.py

```diff
@@ -148,6 +161,13 @@ def create_user(self, claims):
         username = claims.get("preferred_username")
         email = claims.get("email", "")  # Email is optional
         # We create the user based on the username and optional email fields.
+
+        # If a user already exists with this username
+        # change that user's username
+        for user in self.UserModel.objects.filter(username=username):
+            user.username = user.profile.site.pk + '_' + username
+            user.save()
```

Comment:
I think we should only do this if the user that we're clashing with doesn't have an `auth_service_uuid`.
And if they do then we need to raise an error because there's something very wrong.